doc/service_config.md

```diff
@@ -13,12 +13,20 @@ The service config is a JSON string of the following form:
 ```
 {
   // Load balancing policy name.
-  // Supported values are 'round_robin' and 'grpclb'.
-  // Optional; if unset, the default behavior is pick the first available
-  // backend.
-  // Note that if the resolver returns only balancer addresses and no
-  // backend addresses, gRPC will always use the 'grpclb' policy,
-  // regardless of what this field is set to.
+  // Currently, the only selectable client-side policy provided with gRPC
+  // is 'round_robin', but third parties may add their own policies.
+  // This field is optional; if unset, the default behavior is to pick
+  // the first available backend.
+  // If the policy name is set via the client API, that value overrides
+  // the value specified here.
+  //
+  // Note that if the resolver returns at least one balancer address (as
+  // opposed to backend addresses), gRPC will use grpclb (see
+  // https://github.com/grpc/grpc/blob/master/doc/load-balancing.md),
+  // regardless of what LB policy is requested either here or via the
+  // client API.  However, if the resolver returns backend addresses as
+  // well as balancer addresses, the client may fall back to the requested
+  // policy if it is unable to reach any of the grpclb load balancers.
```

Comment:
> Giving the pick_first policy a name here would still allow people to set it explicitly, which is exactly what Abhishek wanted to avoid. 

Is the intention that we don't give people the option to stick to pick_first behavior?

> However, in cases where grpclb is intentionally not being used, I think that's misleading, because we never tried to use grpclb in the first place.

In contrast, I find the current config language misleading, because the option doesn't have any notion of "fallback" or "secondary" but is in fact **used only when the conditions for the primary choice is not met**. The current spec is disguising the primary/secondary relation as some special-case rules. IMHO, specs should avoid special cases whenever possible, i.e., if they can be instead expressed by a more generalized rule. If "fallback" doesn't sound right to you, how about "secondary"?